File: 백준/Silver/2485. 가로수/가로수.py
import sys

a = int(sys.stdin.readline())
b = int(sys.stdin.readline())
arr = []

def GCD(a,b):
    if(a%b==0):
        return b
    return GCD(b,a%b)
  
for i in range(a-1):
    c = int(sys.stdin.readline())
    arr.append(c-b)
    b = c

d = arr[0]
for i in range(1,len(arr)):
    d = GCD(arr[i],d)
  
sum = 0
for x in arr:
    sum += (x//d) - 1
print(sum)

File: 백준/Silver/2485. 가로수/가로수.cc
#include <iostream>
#include <vector>

using namespace std;

int GCD(int a, int b);

int main() {
    int N;
    cin >> N;
    int a;
    cin >> a;
    vector<int> v;
    int b;
    for (int i = 0; i < N-1; i++) {
        cin >> b;
        v.push_back(b - a);
        a = b;
    }
    
    int g = v[0];
    for (int i = 1; i < N-1; i++) {
        g = GCD(g,v[i]);
    }
    
    int sum = 0;
    for (int i = 0; i < v.size(); i++) {
        sum += (v[i]/g) - 1;
    }
    
    cout << sum;
}

int GCD(int a, int b) {
    if(a%b==0) {
        return b;
    } return GCD(b, a%b);
}